client/jvm/src/main/scala/org/http4s/client/JavaNetClientBuilder.scala

```diff
@@ -56,7 +55,7 @@ sealed abstract class JavaNetClientBuilder[F[_]] private (
     val proxy: Option[Proxy],
     val hostnameVerifier: Option[HostnameVerifier],
     val sslSocketFactory: Option[SSLSocketFactory],
-)(implicit protected val F: Async[F])
+)(implicit protected val F: Sync[F])
```

Comment:
Actually I missed that this is a `protected val` in a `sealed` class. So we can make this change on 0.23 actually, we'll just have to filter it.